### PR TITLE
Throw KustoServiceQueryError when Kusto service result includes OneApiError(s)

### PR DESCRIPTION
**Pull request by yihezkel**
### Changed
Minor enhancements regarding reuse, added tests

### Fixed
Throw KustoServiceQueryError when Kusto service result includes OneApiError(s)

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoOperationResult.java [hunk 1]
```diff
@@ -18,6 +18,8 @@
 import java.lang.invoke.MethodHandles;
 import java.util.*;
 
+import static com.microsoft.azure.kusto.data.KustoResultSetTable.EXCEPTIONS_MESSAGE;
```
Comment (ohadbitt) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoOperationResult.java:21]: please don't import static

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoOperationResult.java [hunk 2]
```diff
@@ -122,7 +128,24 @@ private void createFromV2Response(String response) throws KustoServiceQueryError
             if (jsonNode.isArray()) {
                 jsonArray = (ArrayNode) jsonNode;
                 for (JsonNode node : jsonArray) {
-                    if (node.has(FRAME_TYPE_PROPERTY_NAME) && node.get(FRAME_TYPE_PROPERTY_NAME).asText().equals(DATA_TABLE_FRAME_TYPE_PROPERTY_NAME)) {
+                    if (node.has(FRAME_TYPE_PROPERTY_NAME) && node.get(FRAME_TYPE_PROPERTY_NAME).asText().equals(DATA_SET_COMPLETION_FRAME_TYPE_PROPERTY_VALUE)
```
Comment (ohadbitt) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoOperationResult.java:131]: Can do this once
if (node.has(FRAME_TYPE_PROPERTY_NAME) TableType = node.get(FRAME_TYPE_PROPERTY_NAME)

**Review by ohadbitt — APPROVED** [review]
Wow did not know that is an option .. is it new ?

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoOperationResult.java (hunk 1 above)
Comment (yihezkel) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoOperationResult.java:21]: Even for constants? Why not?

**Comment by yihezkel** [comment]
> Looking good Wow did not know that is an option .. is it new ?

I don't think it's new. Also, at least some consumers already throw an exception if the query response object includes a `OneApiError` - C# SDK (though they implement it a little differently) and Kusto Explorer. My next step is to check the other SDKs.

**Pull request merged**